datareduction/datareductionoperator.cpp

```diff
@@ -1436,6 +1436,136 @@ namespace DRO {
       return true;
    }
 
+   // Precipitation directional differential number flux
+   VariablePrecipitationDiffFlux::VariablePrecipitationDiffFlux(cuint _popID): DataReductionOperatorHasParameters(),popID(_popID) {
+      popName = getObjectWrapper().particleSpecies[popID].name;
+      cosAngle = cos(10.*M_PI/180.0); // cosine of fixed loss cone angle
+
+      emin = getObjectWrapper().particleSpecies[popID].emin;    // keV
+      emax = getObjectWrapper().particleSpecies[popID].emax;  // keV
+      nChannels = getObjectWrapper().particleSpecies[popID].nChannels; // number of energy channels, logarithmically spaced between emin and emax
+      for (int i=0; i<nChannels; i++){
+         channels.push_back(emin * pow(emax/emin,float(i)/(nChannels-1)));
+      }
+   }
+   VariablePrecipitationDiffFlux::~VariablePrecipitationDiffFlux() { }
+   
+   std::string VariablePrecipitationDiffFlux::getName() const {return popName + "/PrecipitationDiffFlux";}
+   
+   bool VariablePrecipitationDiffFlux::getDataVectorInfo(std::string& dataType,unsigned int& dataSize,unsigned int& vectorSize) const {
+      dataType = "float";
+      dataSize =  sizeof(Real);
+      vectorSize = nChannels; //Number of energy channels
+      return true;
+   }
+   
+   bool VariablePrecipitationDiffFlux::reduceData(const SpatialCell* cell,char* buffer) {
+
+      dataDiffFlux.assign(nChannels,0.0);
+
+      std::vector<Real> sumWeights(nChannels,0.0);
+
+      std::array<Real,3> B;
+      B[0] = cell->parameters[CellParams::PERBXVOL] +  cell->parameters[CellParams::BGBXVOL];
```

Comment:
Will this conflict with #405 or subsequent coding by @galfthan? In what order should we merge?

Reply:
It will, since this should be changed to overall BVOL then. But I would actually suggest to merge this one first, because it's the smaller change